### PR TITLE
Enforce valid git branch names

### LINKED ISSUE #1724: bug: Strip leading or trailing dash for groupSlug's generated from groupName
#### This is a:

* [X] Bug report (non-security related)
* [ ] Feature request
* [ ] I'm not sure which of those it is

#### I'm using:

* [ ] The Renovate GitHub App
* [X] Self-hosted GitHub
* [ ] Self-hosted GitLab
* [ ] Self-hosted VSTS

#### Please describe the issue:
Consider this config:
```
  "packageRules": [
    {
      "groupName": "@myorg",
      "packagePatterns": [
        "^@myorg"
      ]
    }
  ]
```

When renovate generates the groupSlug/branch name based on this `groupName`; the resulting branch name is "`renovate/-myorg`" (note leading slash)

Same appears to happen for trailing dash. I'm not sure if this is a valid branch name (may be). But, in context of what renovate's intent is - seems like a bug to me.

I can see this is handled [here](https://github.com/renovateapp/renovate/blob/master/lib/workers/repository/updates/branchify.js#L31) - but I wonder if maybe using something like [this](https://www.npmjs.com/package/slugify) would be a better approach?

### PR DESCRIPTION
**Pull request by hulkish**
Fixes #1724

### CODE REVIEW

File: lib/workers/repository/updates/branchify.js [hunk 1]
```diff
@@ -23,12 +25,10 @@ function branchifyUpgrades(config) {
...
           `Dependency ${upgrade.depName} is part of group ${upgrade.groupName}`
         );
-        upgrade.groupSlug =
-          upgrade.groupSlug ||
-          upgrade.groupName
-            .toString()
-            .toLowerCase()
-            .replace(/[^a-z0-9+]+/g, '-');
+        upgrade.groupSlug = slugify(upgrade.groupSlug || upgrade.groupName, {
+          remove: /^-|\.\.|\/$/g,
```
Comment (rarkins) [inline lib/workers/repository/updates/branchify.js]: This pattern needs to be explained/documented

Comment (hulkish) [reply lib/workers/repository/updates/branchify.js]: Removing this actually, letting this be handled via `clean-git-ref`

**Comment by rarkins** [comment]
Thanks for the PR!

This needs some tests though to verify the new functionality. Any change to branch names needs to be checked carefully as it can disturb a lot of users

**Comment by hulkish** [comment]
@rarkins added some tests. I believe I've covered [all of these](https://github.com/renovateapp/renovate/issues/1724#issuecomment-376567029). 

Let me know if you can think of more I should add. I'm guessing there's probably some regressions for current behavior to add here - but you would know that better than I.

**Comment by hulkish** [comment]
Also, This doesn't actually strip leading/trailing slashes (because they are actually valid branch names)... however i suppose it makes sense for renovate to do so. What do u think?

**Comment by rarkins** [comment]
If someone really wants them then let’s let them. In your case the `@` shall remain instead right?

File: yarn.lock [hunk 2]
```diff
@@ -947,6 +947,10 @@ class-utils@^0.3.5:
     isobject "^3.0.0"
     static-extend "^0.1.1"
 
+clean-git-ref@1.0.3:
+  version "1.0.3"
+  resolved "https://repo.artifacts.weather.com/api/npm/web-decoupling-npm-virtual/clean-git-ref/-/clean-git-ref-1.0.3.tgz#5325dc839eab01c974ae0e97f5734782750f88ec"
```
Comment (rarkins) [inline yarn.lock]: lol @hulkish

**Comment by hulkish** [comment]
Yea - `@` is valid

**Comment by hulkish** [comment]
@rarkins ok fixed the yarn.lock, lol.

**Comment by rarkins** [comment]
Needs `yarn prettier` run locally and the change pushed

**Pull request merged**